lib/analyze/collect/npm.js

```diff
@@ -30,16 +30,9 @@ function fetchDownloads(name) {
         timeout: 15000,
         retries: gotRetries,
     })
-    // Their API does not respect the HTTP status codes.. they respond 200 OK even if the request failed.
-    .then((response) => {
-        if (response.body.error) {
-            throw new Error(response.body.error);
-        }
-
-        return response.body.downloads;
-    })
-    // Check if the error is "no stats" and return an empty array
-    .catch((err) => /no stats for this package/i.test(err.message), () => [])
+    .then((res) => res.body.downloads)
+    // Check if there is no stats yet
+    .catch({ statusCode: 404 }, () => [])
```

Comment:
got throws an error that has `statusCode` and bluebird allows to do `catch` this way.